pkg/controller/kabaneroplatform/featured_stacks_test.go

```diff
@@ -509,3 +568,330 @@ func TestResolveFeaturedStacksTwoRepositories(t *testing.T) {
 		t.Fatal(fmt.Sprintf("Expected two versions of nodejs stack, but found %v: %v", len(nodejsStackVersions), nodejsStackVersions))
 	}
 }
+
+// Tests that if an existing stack version with an non-active desired state matches the name/version of a stack in the index, the
+// data associated with the index version overrides the existing version.
+// Tests that a stack version with an unset desired state is removed if it is not found in the index being deployed.
+// Tests that if an existing stack version with an active state is not found in the index being deployed, the existing version
+// remains untouched.
+func TestResolveFeaturedStacksCleanupMatchingStackVersionWithNonActiveState(t *testing.T) {
+	stack := stackResource.DeepCopy()
+	stack.Spec.Versions[0].DesiredState = "NotAValidState"
```

Comment:
This test probably gets simpler if we treat all non-null desiredState values the same.